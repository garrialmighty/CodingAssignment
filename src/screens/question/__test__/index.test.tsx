import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import QuestionScreen from '../';

jest.mock('src/hooks/useAppDispatch', () => jest.fn());

const mockBack = jest.fn();
const props = {
  navigation: {
    goBack: mockBack,
    setOptions: jest.fn(),
  },
};

jest.mock('../hooks/useIndexer', () => {
  const {DefaultCategories} = require('src/data/categories');
  return () => ({
    questionIndex: 1,
    totalQuestions: 1,
    question: 'Mock Question',
    answer: DefaultCategories[0].items[0].answer,
  });
});

describe('Screen: Landing', () => {
  beforeEach(() => {
    mockBack.mockClear();
  });

  it('should render properly', () => {
    const snapshot = render(<QuestionScreen {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('navigates back when tapping Next on the last question', () => {
    const {getByTestId} = render(<QuestionScreen {...props} />);
    fireEvent.press(getByTestId('category-button-Skip'));
    expect(mockBack).toHaveBeenCalled();
  });
});
