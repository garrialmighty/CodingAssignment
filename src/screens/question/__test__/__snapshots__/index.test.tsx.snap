// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen: Landing should render properly 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
      "justifyContent": "space-between",
      "marginHorizontal": "5%",
    }
  }
>
  <Text
    style={
      {
        "backgroundColor": "#dce7f9",
        "borderColor": "#a2b4cc",
        "borderWidth": 1,
        "fontSize": 50,
        "textAlign": "center",
      }
    }
  >
    1/1
  </Text>
  <View
    style={
      {
        "flex": 0.8,
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        {
          "alignContent": "center",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "rowGap": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-S"
      >
        <Text>
          S
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-I"
      >
        <Text>
          I
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-N"
      >
        <Text>
          N
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-G"
      >
        <Text>
          G
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-A"
      >
        <Text>
          A
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-P"
      >
        <Text>
          P
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-O"
      >
        <Text>
          O
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-R"
      >
        <Text>
          R
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-E"
      >
        <Text>
          E
        </Text>
      </View>
    </View>
    <Text
      style={
        {
          "fontSize": 30,
          "textAlign": "center",
        }
      }
    >
      Mock Question
    </Text>
    <View
      style={
        {
          "alignContent": "center",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "rowGap": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-S"
      >
        <Text>
          S
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-I"
      >
        <Text>
          I
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-N"
      >
        <Text>
          N
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-G"
      >
        <Text>
          G
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-A"
      >
        <Text>
          A
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-P"
      >
        <Text>
          P
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-O"
      >
        <Text>
          O
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-R"
      >
        <Text>
          R
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginRight": 20,
            "opacity": 1,
            "width": 45,
          }
        }
        testID="category-button-E"
      >
        <Text>
          E
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#f0cfcd",
        "borderColor": "#b88782",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 45,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="category-button-Start"
  >
    <Text>
      Start
    </Text>
  </View>
</RCTSafeAreaView>
`;
